mirrord/layer/src/debugger_ports.rs

```diff
@@ -175,26 +276,93 @@ impl DebuggerType {
                 None
             }
         })
+        .collect::<Vec<u16>>();
+
+        let next_type = match self {
+            // node may require several rounds of port ignoring when used in the VSCode plugin
+            DebuggerType::NodeInspector => Some(DebuggerType::NodeInspector),
+            _ => None,
+        };
+        (ports, next_type)
     }
 }
 
 /// Local ports used by the debugger running the process.
 /// These should be ignored by the layer.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub enum DebuggerPorts {
-    Detected(u16),
+    Single(u16),
     FixedRange(RangeInclusive<u16>),
+    Combination(Vec<DebuggerPorts>),
     None,
 }
 
+impl FromStr for DebuggerPorts {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // string looks like 'port1,port2,port3-portx,porty-portz'
+        let mut vec = vec![];
+        s.split(',')
+            .for_each(|s| {
+                let chunks = s
+                    .split('-')
+                    .map(u16::from_str)
+                    .collect::<Result<Vec<_>, _>>()
+                    .inspect_err(|e| error!(
+                        "Failed to decode debugger port range from {} env variable: {}",
+                        MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
+                        e
+                    ))
+                    .ok().unwrap_or_default();
+                match *chunks.as_slice() {
+                    [p] => vec.push(Self::Single(p)),
+                    [p1, p2] if p1 <= p2 => vec.push(Self::FixedRange(p1..=p2)),
+                    _ => {
+                        error!(
+                            "Failed to decode debugger ports from {} env variable: expected a port or range of ports",
+                            MIRRORD_IGNORE_DEBUGGER_PORTS_ENV,
+                        );
+                    },
+                };
+            });
+        if !vec.is_empty() {
+            Ok(Self::Combination(vec))
+        } else {
+            Err(format!("couldn't parse ports from string: {s}"))
```

Comment:
```suggestion
            Ok(Self::None)
```
and `<DebuggerPorts as FromStr>::Err = std::convert::Infallible`